moshi/src/main/java/com/squareup/moshi/ClassJsonAdapter.java

```diff
@@ -45,8 +46,10 @@
   public static final JsonAdapter.Factory FACTORY = new JsonAdapter.Factory() {
     @Override public @Nullable JsonAdapter<?> create(
         Type type, Set<? extends Annotation> annotations, Moshi moshi) {
-      if (!(type instanceof Class)) return null;
-      Class<?> rawType = (Class<?>) type;
+      if (!(type instanceof Class || type instanceof ParameterizedType)) {
```

Comment:
I find this kind of expression difficult to read; like a double negative. For whatever reason (maybe just me) I find the contrapositive easier.

```
if (!(type instanceof Class) && !(type instanceof ParameterizedType)) {
  return null;
}
```

Reply:
totally. thanks for point that out. #444

Reply:
totally. thanks for pointing that out. #444